### PR TITLE
FOG._wait_for_ready: Tolerate ConnectionResetError

### PR DESCRIPTION
**Pull request by zmc**
Looks like this was missed during PR submission/review in 8f8d058

Signed-off-by: Zack Cerza <zack@redhat.com>

### CODE REVIEW

**Comment by zmc** [comment]
Initially I was testing this by reimaging a single machine in a loop, but I never hit the error; likely because the lab was relatively idle. Next, I tried locking/reimaging 50 smithi machines at once. It succeeded where it wouldn't have without this PR:
```
(virtualenv) zack@teuthology:~/teuthology$ grep -i 'connection reset' bigrt.log
2023-06-30 16:54:49,200.200 ERROR:paramiko.transport:Socket exception: Connection reset by peer (104)
2023-06-30 16:54:49,200.200 ERROR:teuthology.provision.fog:[Errno 104] Connection reset by peer on smithi026
2023-06-30 16:55:34,256.256 ERROR:paramiko.transport:Socket exception: Connection reset by peer (104)
2023-06-30 16:55:34,257.257 ERROR:teuthology.provision.fog:[Errno 104] Connection reset by peer on smithi092
2023-06-30 16:56:11,120.120 ERROR:paramiko.transport:Socket exception: Connection reset by peer (104)
2023-06-30 16:56:11,120.120 ERROR:teuthology.provision.fog:[Errno 104] Connection reset by peer on smithi002
2023-06-30 16:56:41,840.840 ERROR:paramiko.transport:Socket exception: Connection reset by peer (104)
2023-06-30 16:56:41,840.840 ERROR:teuthology.provision.fog:[Errno 104] Connection reset by peer on smithi097
2023-06-30 16:59:09,296.296 ERROR:paramiko.transport:Socket exception: Connection reset by peer (104)
2023-06-30 16:59:09,297.297 ERROR:teuthology.provision.fog:[Errno 104] Connection reset by peer on smithi096
2023-06-30 17:00:23,024.024 ERROR:paramiko.transport:Socket exception: Connection reset by peer (104)
2023-06-30 17:00:23,025.025 ERROR:teuthology.provision.fog:[Errno 104] Connection reset by peer on smithi023
```

The paramiko log message is not new, but the message generated by our `fog` module is logged as part of the retry mechanism added by this PR.

File: teuthology/contextutil.py [hunk 1]
```diff
@@ -139,7 +139,6 @@ def __call__(self):
         if self.timeout > 0:
             self.sleep_current = min(self.timeout - self.total_seconds, self.sleep_current)
         self.total_seconds += self.sleep_current
-        print(self.total_seconds, self.sleep_current)
```
Comment (ljflores) [inline teuthology/contextutil.py:142]: Why was this piece removed?

File: teuthology/provision/fog.py [hunk 2]
```diff
@@ -292,7 +292,14 @@ def _wait_for_ready(self):
         sentinel_file = config.fog.get('sentinel_file', None)
         if sentinel_file:
             cmd = "while [ ! -e '%s' ]; do sleep 5; done" % sentinel_file
-            self.remote.run(args=cmd, timeout=600)
+            action = f"wait for sentinel on {self.shortname}"
+            with safe_while(action=action, timeout=1800, increment=3) as proceed:
+                while proceed():
+                    try:
+                        self.remote.run(args=cmd, timeout=600)
```
Comment (ljflores) [inline teuthology/provision/fog.py:299]: How did you decide on these timeouts?

Overall LGTM, my comments are mainly trying to understand. Let's test it with the nop suite.

**Review by ljflores — DISMISSED** [review]

File: teuthology/provision/fog.py (hunk 2 above)
Comment (jdurgin) [inline teuthology/provision/fog.py:299]: nice catch! wasn't thinking the steps after connection could have similar issues

File: teuthology/contextutil.py (hunk 1 above)
Comment (zmc) [reply teuthology/contextutil.py:142]: It was what appears to be a debug statement left over from development in the commit mentioned in my commit message. Results in random integers being printed to the console from who-knows-where

**Review by jdurgin — DISMISSED** [review]

File: teuthology/provision/fog.py (hunk 2 above)
Comment (zmc) [reply teuthology/provision/fog.py:299]: The 1800 value came out of nowhere, really; accepting suggestions :)

Comment (ljflores) [reply teuthology/provision/fog.py:299]: That's fine to start, was just curious how you decided on the value

**Comment by ljflores** [comment]
Test run:
https://pulpito.ceph.com/zack-2023-06-30_17:59:19-teuthology:nop-main-distro-default-smithi/

**Comment by zmc** [comment]
First test run did quite well (47/50 passed):
https://pulpito.ceph.com/zack-2023-06-30_17:59:19-teuthology:nop-main-distro-default-smithi/
But I think I can do a bit better by rewriting `Remote.reconnect()`:
https://pulpito.ceph.com/zack-2023-06-30_18:31:23-teuthology:nop-main-distro-default-smithi/

**Review by ljflores — APPROVED** [review]
LGTM!

**Pull request merged**